Gems/AWSMetrics/Code/Source/IdentityProvider.cpp

```diff
@@ -15,18 +15,32 @@
 
 namespace AWSMetrics
 {
+    constexpr const char* EngineVersionJsonKeyFileFormat1 = "O3DEVersion";
+    constexpr const char* EngineVersionJsonKeyFileFormat2 = "display_version";
+
     AZStd::unique_ptr<IdentityProvider> IdentityProvider::CreateIdentityProvider()
     {
         return AZStd::make_unique<DefaultClientIdProvider>(GetEngineVersion());
     }
 
     AZStd::string IdentityProvider::GetEngineVersion()
     {
-        constexpr auto engineVersionKey = AZ::SettingsRegistryInterface::FixedValueString(AZ::SettingsRegistryMergeUtils::EngineSettingsRootKey) + "/" + EngineVersionJsonKey;
+        constexpr auto rootKey = AZ::SettingsRegistryInterface::FixedValueString(AZ::SettingsRegistryMergeUtils::EngineSettingsRootKey);
+        constexpr auto engineVersionKeyFileFormat1 = rootKey + "/" + EngineVersionJsonKeyFileFormat1;
+        constexpr auto engineVersionKeyFileFormat2 = rootKey + "/" + EngineVersionJsonKeyFileFormat2;
         AZStd::string engineVersion;
-        if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr && settingsRegistry->Get(engineVersion, engineVersionKey))
+
+        if(auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
         {
-            return engineVersion;
+            // prefer latest file format
+            if(settingsRegistry->Get(engineVersion, engineVersionKeyFileFormat2))
+            {
+                return engineVersion;
+            }
+            else if(settingsRegistry->Get(engineVersion, engineVersionKeyFileFormat1))
+            {
+                return engineVersion;
+            }
```

Comment:
It seems like we can just add a GetEngineDisplayVersion function to the AzCore [Utils.h/cpp](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L95-L100) file instead of using ad-hoc reading of the setting.